### PR TITLE
feat: implement upload collection

### PR DESCRIPTION
**Pull request by mikerodonnell89**
## Related Issue
Closes SAP/fundamental-styles #2578 

## Description
This PR introduces the upload collection for fundamental-styles, which is essentially a byline list with some modifications.

## Screenshots
Screen Shot 2021-07-26 at 11 50 05 AM
Screen Shot 2021-07-26 at 11 50 00 AM

#### Please check whether the PR fulfills the following requirements

1. The output matches the design specs
- [x] All values are in `rem`
- [x] Text elements follow the truncation rules
- [x] hover state of the element follow design spec
- [x] focus state of the element follow design spec
- [x] active state of the element follow design spec
- [x] selected state of the element follow design spec
- [x] selected hover state of the element follow design spec
- [x] pressed state of the element follow design spec
- [x] Responsiveness rules - the component has modifier classes for all breakpoints
- [x] Includes Compact/Cosy/Tablet design
- [x] RTL support
2. The code follows fundamental-styles code standards and style
- [x] only one top level `fd-*` class is used in the file
- [x] BEM naming convention is used
- [x] Mixins are used for repeatable code (`fd-rtl`, `fd-ellipsis`, `fd-flex`, `fd-selected`, `fd-focus`, ect.)
- [x] A11y support - keyboard support, screenreader support, proper ARIA attributes, etc.
- [x] `fd-reset()` mixin is applied to all elements
- [x] Variables are used, if some value is used more than twice.
- [x] Checked if current components can be reused, instead of having new code.
3. Testing
- [x] tested Storybook examples with "CSS Resources" `normalize` option 
- [x] tested Storybook examples with "CSS Resources" `unnormalize` option 
- [x] Verified all styles in IE11
- [x] Updated tests
- [x] last commit message should have `[ci visual]` so it can trigger chromatic visual regression (e.g. `test: run chromatic visual regression [ci visual]`)
4. Documentation
- [x] Storybook documentation has been created/updated
- [x] Breaking Changes [wiki](https://github.com/SAP/fundamental-styles/wiki/Breaking-Changes) has been updated in case of breaking changes.

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
The Avatar should be top aligned.
Specs:
Screen Shot 2021-07-26 at 1 42 32 PM

Fundamental:
Screen Shot 2021-07-26 at 1 43 27 PM

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-07-26 at 1 50 19 PM

Could you add an example like this with more items so we have the Chromatic screenshot and detect changes/bugs in future

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-07-26 at 1 53 47 PM

Is this part of the ticket or we leave it for another phase?

**Comment by InnaAtanasova** [comment]
Screen Shot 2021-07-26 at 1 55 57 PM

I think you are missing this part in the documentation.

File: src/upload-collection.scss [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    &--small {
+      input {
+        width: 75%;
+      }
+    }
+  }
+
+  &__content {
+    flex: 1 !important;
```
[the remark this answers is not in the document]
Comment (mikerodonnell89) [reply src/upload-collection.scss:99]: removed

**Comment by mikerodonnell89** [comment]
> Screen Shot 2021-07-26 at 1 53 47 PM
> 
> Is this part of the ticket or we leave it for another phase?

So this is a tricky one.  The specs want us to always display the extension.  There are several ways to truncate some text in the middle of the string rather than at the end.  The only purely CSS/HTML ways I am aware of have you move the filename and the extension into different elements.  This works but then you essentially have two different links and it's visually apparent depending on whether you're highlighting (or clicking) the extension or the filename itself.  So I think we should use JavaScript to do this.  Open to discussion / more ideas, lets talk in scrum

**Comment by InnaAtanasova** [comment]
Please trigger Chromatic

**Review by InnaAtanasova — APPROVED** [review]

**Pull request merged**